### DISCUSSION TITLE
Cannot backup/restore PV, PVC only

### DISCUSSION THREAD

**Discussion by JDHawken** [vmware-tanzu/velero#6460] [Community support Q&A]
Hi there,
I got a problem with Velero.
I want to migrate data (inside PVC) from application V1 to V2 as you can see on this drawing I did :
![index](https://github.com/vmware-tanzu/velero/assets/121531022/9aa478c7-8d99-4185-9cea-e2d64cfd3957)

But impossible to restore data, when I do my velero restore with the following command, my pvc are empty and I didn't understand why, 
`velero restore create --from-backup my-backup --include-resources pvc,pv --include-namespaces myappv2`

For exemple, the data I have inside myappv1 in ElasticSearchData : 
`$ ls /bitnami/elasticsearch/data/nodes/0/indices/
FtMKs5RLRMSVBhvdm8N93g  gtNIwzqKSsebl5q8MXa8BQ  rpCRBWlBRAu2j45BVzaKJQ`

The data I have inside myappv2 in ElasticSearchData after velero restore (include pvc, pv only) : 
`$ ls /bitnami/elasticsearch/data/nodes/0/indices/
ls: cannot access '/bitnami/elasticsearch/data/nodes/0/indices/': No such file or directory
$ ls /bitnami/elasticsearch/data/nodes/0/
_state  node.lock`

Strangely, when I backup everything on Cluster01 like this :
`velero backup create backup-myapp01 --include-namespaces myapp01`

And redeploy including this : 
`velero restore create --from-backup backup-myapp01 --include-resources pvc,pv`

It doesn't work, but if I restore everything without "--include-ressources pvc,pv", I find again my data, I conclude that my data is in backup velero, but why I can't restore only them with include-ressources ?
Thanks in advance.

**Comment by draghuram** [comment]
It is interesting that we had similar report from a CloudCasa user very recently (CloudCasa uses Velero but implements some more functionality on the top). PVC data is restored in the context of pods and since you didn't include pods, no data is restored. Can you include "pods" in the resource types, in addition to "pvc"? Note that this is only an issue for file system backups. If you have snapshot backups, you can easily restore PVCs alone to an alternate namespace and then copy over data.

What version of Velero are you using BTW?
